shared/src/utils/type-state-selector-map.ts

```diff
@@ -0,0 +1,15 @@
+export const typeSelectorMap = {
```

Comment:
I don't really like this name as one could easily confuse this with the whole selector-concept from Redux that we use too.

What about `ElementTypePluralMap`?

Reply:
Yes, the name isn't that nice. It was simply the first name that came to my mind when I first implemented this to test the typings.

I've changed it according to your suggestion here and renamed the derived types (in `shared/src/store/action-reducers/exercise.ts` and `shared/src/store/action-reducers/utils/spatial-elements.ts`), too